erpnext/hr/doctype/staffing_plan/staffing_plan.js

```diff
@@ -25,69 +25,63 @@ frappe.ui.form.on('Staffing Plan', {
 				}
 			};
 		});
-	}
+	},
 });
 
 frappe.ui.form.on('Staffing Plan Detail', {
 	designation: function(frm, cdt, cdn) {
-		let child = locals[cdt][cdn]
-		if(frm.doc.company && child.designation){
-			frappe.call({
-				"method": "erpnext.hr.doctype.staffing_plan.staffing_plan.get_designation_counts",
-				args: {
-					designation: child.designation,
-					company: frm.doc.company
-				},
-				callback: function (data) {
-					if(data.message){
-						frappe.model.set_value(cdt, cdn, 'current_count', data.message.employee_count);
-						frappe.model.set_value(cdt, cdn, 'current_openings', data.message.job_openings);
-						if (child.number_of_positions < (data.message.employee_count +  data.message.job_openings)){
-							frappe.model.set_value(cdt, cdn, 'number_of_positions', data.message.employee_count +  data.message.job_openings);
-						}
-					}
-					else{ // No employees for this designation
-						frappe.model.set_value(cdt, cdn, 'current_count', 0);
-						frappe.model.set_value(cdt, cdn, 'current_openings', 0);
-					}
-				}
-			});
+		let child = locals[cdt][cdn];
+		if(frm.doc.company && child.designation) {
+			set_number_of_positions(frm, cdt, cdn);
 		}
 	},
 
-	number_of_positions: function(frm, cdt, cdn) {
-		set_vacancies(frm, cdt, cdn);
+	vacancies: function(frm, cdt, cdn) {
+		let child = locals[cdt][cdn];
+		if(child.vacancies < child.current_openings) {
+			frappe.throw(__("Vacancies cannot be lower than the current openings"));
+		}
+		set_number_of_positions(frm, cdt, cdn);
 	},
 
 	current_count: function(frm, cdt, cdn) {
-		set_vacancies(frm, cdt, cdn);
+		set_number_of_positions(frm, cdt, cdn);
 	},
 
 	estimated_cost_per_position: function(frm, cdt, cdn) {
-		let child = locals[cdt][cdn];
 		set_total_estimated_cost(frm, cdt, cdn);
 	}
-
 });
 
-var set_vacancies = function(frm, cdt, cdn) {
-	let child = locals[cdt][cdn]
-	if (child.number_of_positions < (child.current_count + child.current_openings)){
-		frappe.throw(__("Number of positions cannot be less then current count of employees"))
-	}
-
-	if(child.number_of_positions > 0) {
-		frappe.model.set_value(cdt, cdn, 'vacancies', child.number_of_positions - (child.current_count + child.current_openings));
-	}
-	else{
-		frappe.model.set_value(cdt, cdn, 'vacancies', 0);
-	}
-
+var set_number_of_positions = function(frm, cdt, cdn) {
```

Comment:
use `let` instead of `var`

Reply:
let has a limited scope, it cannot be used in this specific case.